DeusEx/Classes/DXRMenuSetupRando.uc

```diff
@@ -377,32 +433,110 @@ function ProcessAction(String actionKey)
         f.speedlevel = GetSliderValue(editSpeedLevel);
         f.dancingpercent = GetSliderValue(editDancingPercent);
 
+        if( sgamemode == "Original Story" )
+            f.gamemode = 0;
+        else if( sgamemode == "Original Story, Rearranged Levels (Beta)" )
+            f.gamemode = 1;
+
         if( keys == "Off" ) f.keysrando = 0;
         else if( keys == "Dumb" ) f.keysrando = 2;
         else if( keys == "On" ) f.keysrando = 4;
         else if( keys == "Copy" ) f.keysrando = 3;
         else if( keys == "Smart" ) f.keysrando = 4;
 
-        if( doors == "Unchanged" ) {
+        undefeatabledoors = 1*256;
+        alldoors = 2*256;
+        keyonlydoors = 3*256;
+        highlightabledoors = 4*256;
+        doormutuallyinclusive = 1;
+        doorindependent = 2;
+        doormutuallyexclusive = 3;
+
+        if( doors == "Doors Unchanged" ) {

```

Comment:
maybe I should rewrite all these if statements, I could make this check for keywords like "Undefeatable Doors" and "Some Doors" pretty easily